R/epichains.R

```diff
@@ -136,13 +138,33 @@ epichains_summary <- function(chains_summary,
                               index_cases,
                               statistic,
                               offspring_dist,
-                              stat_max) {
-  # Check that inputs are well specified
-  checkmate::assert_vector(chains_summary)
-  checkmate::assert_integerish(index_cases, null.ok = TRUE)
-  checkmate::assert_character(statistic)
+                              stat_max = Inf) {
+  # chain_summary can sometimes contain infinite values, so check
+  # that it's all infinite OR the non-infinite values are integerish.
+  # If it doesn't contain infinite values, then they are all integerish.
+  if (any(is.infinite(chains_summary))) {
+    checkmate::assert(
+      all(is.infinite(chains_summary)),
+      checkmate::check_integerish(
+        chains_summary[!is.infinite(chains_summary)],
+        lower = 0,
+        any.missing = FALSE
+      )
+    )
+  } else {
+    checkmate::assert_integerish(
+      chains_summary,
+      len = index_cases,
+      lower = 0
+    )
+  }
```

Comment:
Could this be replaced by the code below, which checks that any non-infinite values are integerish? Is this what you intended?

``` r
v1 = c(1, 2, Inf)
v2 = c(Inf, Inf)
f = function(x) {
  checkmate::assert_integerish(
    x[is.finite(x)], lower = 1
  ) 
}

f(v1)
f(v2)
f(c(1.1, Inf, 1))
#> Error in f(c(1.1, Inf, 1)): Assertion on 'x[is.finite(x)]' failed: Must be of type 'integerish', but element 1 is not close to an integer.
```

<sup>Created on 2024-03-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Super! Thanks.